### PR TITLE
chore: codacity

### PR DESCRIPTION
**Pull request by henryiii**
@phlptp, you might want to review the last commit here, as some of those are touching things you've worked on.

- docs: fix some Codacity recommendations
- chore: update copyright year
- style: more codacity fixes
- style: fix issues reported by Codacity

### CODE REVIEW

File: include/CLI/App.hpp [hunk 1]
```diff
@@ -1763,7 +1763,7 @@ class App {
         if(name_.empty()) {
             return std::string("[Option Group: ") + get_group() + "]";
         }
-        if(aliases_.empty() || !with_aliases || aliases_.empty()) {
+        if(aliases_.empty() || !with_aliases) {
```
Comment (henryiii) [inline include/CLI/App.hpp:1766]: Assuming this was not trying to check something else?

File: include/CLI/Validators.hpp [hunk 2]
```diff
@@ -1113,7 +1113,6 @@ inline std::pair<std::string, std::string> split_program_name(std::string comman
                     esp = end + 1;
                     if(embeddedQuote) {
                         vals.first = find_and_replace(vals.first, std::string("\\") + keyChar, std::string(1, keyChar));
-                        embeddedQuote = false;
```
Comment (henryiii) [inline include/CLI/Validators.hpp:1116]: This is never used. The variable gets reinitialized to false after this ends.

File: tests/AppTest.cpp [hunk 3]
```diff
@@ -1933,7 +1933,6 @@ TEST_CASE_METHOD(TApp, "AllowExtrasArgModify", "[app]") {
     app.allow_extras();
     app.add_option("-f", v2);
     args = {"27", "-f", "45", "-x"};
-    auto cargs = args;
```
Comment (henryiii) [inline tests/AppTest.cpp:1936]: Unused, hopefully this was not supposed to be used later for a missing test?

File: include/CLI/App.hpp (hunk 1 above)
Comment (phlptp) [reply include/CLI/App.hpp:1766]: I suspect it was copied and ended up duplicated condition

File: tests/AppTest.cpp (hunk 3 above)
Comment (phlptp) [reply tests/AppTest.cpp:1936]: Seeing how the method names is argModify I could see that being potentially useful, but it look like the test didn't end up using it and just making the check with a constructed vector.

File: tests/HelpersTest.cpp [hunk 4]
```diff
@@ -998,6 +998,10 @@ TEST_CASE("Types: TypeName", "[helpers]") {
     CHECK(umapName == "[INT,[TEXT,FLOAT]]");
 
     vclass = CLI::detail::classify_object<std::atomic<int>>::value;
+    CHECK(CLI::detail::object_category::wrapper_value == vclass);
+
+    std::string atomic_name = CLI::detail::type_name<std::atomic<int>>();
+    CHECK(atomic_name == "INT");
```
Comment (henryiii) [inline tests/HelpersTest.cpp]: @phlptp This actually reports `CLI::detail::object_category::other` and "TEXT" on some compilers (older clang and GCC). Any quick thoughts on if that is easily fixable / important?

Comment (phlptp) [reply tests/HelpersTest.cpp]: Things got a little weird with older compilers and atomics and the template checks.   I think it still works when actually used just in a roundabout fashion through the stream operators and I am not sure the effort to fix the template names for those compilers is worth it.

Comment (henryiii) [reply tests/HelpersTest.cpp]: Okay, we can just check for both then.

**Pull request merged**